odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/DataSourceIngestionServiceImpl.java

```diff
@@ -1,76 +1,115 @@
-
-        final List<DataSource> createdDataSources = Stream.of(
-                createdDtos,
-                deletedSplitDataSources.get(true),
-                deletedSplitDataSources.get(false))
-            .flatMap(Collection::stream)
-            .map(dataSourceIngestionMapper::mapDtoToIngestionModel)
+                        final Flux<DataSourcePojo> createdDataSources =
+                            reactiveDataSourceRepository.bulkCreate(toCreate);
+
+                        return Flux.concat(updatedDataSources, createdDataSources);
+                    })
+                    .map(ds -> new DataSourceDto(ds, c.namespace(), c.tokenDto()))
+                    .map(dataSourceIngestionMapper::mapDtoToIngestionModel);
+            });
+    }
+
+    private List<DataSourcePojo> prepareForUpdate(final List<DataSourcePojo> actual,
+                                                  final List<DataSourcePojo> ingested) {
+        final Map<String, DataSourcePojo> actualDict =
+            actual.stream().collect(toMap(DataSourcePojo::getOddrn, identity()));
+
+        return ingested.stream()
+            .map(i -> {
+                final DataSourcePojo a = actualDict.get(i.getOddrn());
+                if (a == null) {
+                    return null;
+                }
+
+                return new DataSourcePojo(a)
+                    .setName(i.getName())
+                    .setDescription(i.getDescription())
+                    .setUpdatedAt(LocalDateTime.now())
+                    .setIsDeleted(false);
+            })
+            .filter(Objects::nonNull)
             .toList();
+    }
 
-        dataSourceRepository.setTokenFromCollector(oddrns, collector.tokenDto().tokenPojo().getId());
+    private List<DataSourcePojo> prepareForCreate(final List<DataSourcePojo> ingested,
+                                                  final Set<String> excludedOddrns) {
+        return ingested.stream().filter(i -> excludedOddrns.contains(i.getOddrn())).toList();
+    }
 
-        return Mono.just(createdDataSources);
+    private List<DataSourcePojo> mapDataSources(final DataSourceList dataSourceList, final CollectorDto c) {
+        return dataSourceList.getItems().stream()
+            .map(ds -> dataSourceIngestionMapper.mapIngestionModel(
+                ds,
+                defaultFieldIfNull(c.namespace(), NamespacePojo::getId),
+                defaultFieldIfNull(c.tokenDto().tokenPojo(), TokenPojo::getId)
+            ))
+            .toList();
     }
 
-    private List<String> exctractOddrns(final List<DataSourceDto> dtos) {
+    private List<String> extractOddrns(final List<DataSourcePojo> dtos) {
         return dtos.stream()
-            .map(dto -> dto.dataSource().getOddrn())
+            .map(DataSourcePojo::getOddrn)
             .toList();
     }
+
+    private <T, S> S defaultFieldIfNull(final T object, final Function<T, S> mapper) {
```

Comment:
I'm not fully understand why this method is called defaultFieldIfNull
May be it's extractFieldFromNullableObject()?